src/content/drafts/sourcecode-to-tokens-AST.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: post
+title: How a parser translates text into tokens and then into an abstract syntax tree
+teaser: "You will learn how the source code in any proramming language is converted to token and then parsed to generate an AST. The AST can then be traversed to write and build custom rules in ESLint."
+image: 
+authors:
+  - Gayathri Krishnaswamy
+categories:
+  - Blog post
+tags:
+  - Lexical analysis
+  - Syntax analysis
+  - Abstract Syntax Tree
+  - AST
+  - Tokens
+  - ESLint rules
+---
+##  Introduction
+[ESLint](https://eslint.org/docs/latest/) is an open-source pluggable JavaScript linter that uses the Abstract Syntax Tree (AST) of your source code to identify and report patterns found in ECMAScript/JavaScript code. The tool also provides an option to write and configure your own rules to enforce a particular coding style that is relevant only to your project.
+
+In this article, first, we'll see on a high-level how your source code is converted into multiple tokens and is then parsed to produce an AST that represents the syntactic structure of your source code. Next, we'll understand how ESLint leverages the capability of AST and helps you build your own custom rules.
+
+Before we jump into details, let's understand a couple of key terminologies:
+
+- Compiler - A [compiler](https://en.wikipedia.org/wiki/Compiler) is a special program that converts the programming language’s source code into machine-readable code that targets specific operating systems and computer architectures. 
+
+- Parser - A [parser](https://en.wikipedia.org/wiki/Parsing) is a program that is part of the compiler that generates a very abstract representation of your source code, often a tree or a graph.
+
+Now, let's look into the main steps involved in how compilers process the source code:
```

Comment:
Again, because ESLint isn't a compiler, I don't think compilers should be the focus of the post. Parsers should be the focus.

Reply:
Please refer https://github.com/eslint/eslint.org/pull/324#discussion_r953403330